src/protocols/LoRaWAN/LoRaWAN.cpp

```diff
@@ -54,133 +43,193 @@ void LoRaWANNode::setCSMA(uint8_t backoffMax, uint8_t difsSlots, bool enableCSMA
     this->enableCSMA = enableCSMA;
 }
 
+#if !defined(RADIOLIB_EEPROM_UNSUPPORTED)
+void LoRaWANNode::wipe() {
+  Module* mod = this->phyLayer->getMod();
+  mod->hal->wipePersistentStorage();
+}
 
-int16_t LoRaWANNode::restoreOTAA() {
-  int16_t state = this->setPhyProperties();
-  RADIOLIB_ASSERT(state);
+int16_t LoRaWANNode::restore() {
+  // if already joined, ignore
+  if(this->isJoinedFlag) {
+    return(RADIOLIB_ERR_NONE);
+  }
 
-  // check the magic value
   Module* mod = this->phyLayer->getMod();
-  if(mod->hal->getPersistentParameter<uint32_t>(RADIOLIB_PERSISTENT_PARAM_LORAWAN_MAGIC_ID) != RADIOLIB_LORAWAN_MAGIC) {
-    // the magic value is not set, user will have to do perform the join procedure
-    return(RADIOLIB_ERR_NETWORK_NOT_JOINED);
-  }
 
-  // in case of future revisions of NVM, use a version parameter to allow transitioning from one version to another while keeping session alive
-  uint16_t nvm_table_version = mod->hal->getPersistentParameter<uint32_t>(RADIOLIB_PERSISTENT_PARAM_LORAWAN_VERSION_ID);
-  // if (RADIOLIB_PERSISTENT_PARAM_LORAWAN_VERSION > nvm_table_version) {
+  uint8_t nvm_table_version = mod->hal->getPersistentParameter<uint8_t>(RADIOLIB_PERSISTENT_PARAM_LORAWAN_TABLE_VERSION_ID);
+  // if (RADIOLIB_PERSISTENT_PARAM_LORAWAN_TABLE_VERSION > nvm_table_version) {
   //  // set default values for variables that are new or something
   // }
   (void)nvm_table_version;
 
-  // pull all needed information from persistent storage
+  // check the magic value
+  if(mod->hal->getPersistentParameter<uint16_t>(RADIOLIB_PERSISTENT_PARAM_LORAWAN_MAGIC_ID) != RADIOLIB_LORAWAN_MAGIC) {
+    RADIOLIB_DEBUG_PRINTLN("magic id not set (no saved session)");
+    RADIOLIB_DEBUG_PRINTLN("first 16 bytes of NVM:");
+    uint8_t nvmBuff[16];
+    mod->hal->readPersistentStorage(mod->hal->getPersistentAddr(0), nvmBuff, 16);
```

Comment:
This read and the buffer should be guarded by `#if defined(RADIOLIB_DEBUG)`, because out of debug mode, this has no real effect, but may not get optimized away during compilation.